sdv/single_table/base.py

```diff
@@ -276,3 +294,648 @@ def update_transformers(self, column_name_to_transformer):
+                sampled = self._filter_conditions(sampled, conditions, float_rtol)
+
+            num_valid = len(sampled)
+
+            return sampled, num_valid
+
+        else:
+            sampled = pd.DataFrame(index=range(num_rows))
+            sampled = self._data_processor.reverse_transform(sampled)
+            return sampled, num_rows
+
+    def _sample_batch(self, batch_size, max_tries=100,
+                      conditions=None, transformed_conditions=None, float_rtol=0.01,
+                      progress_bar=None, output_file_path=None):
+        """Sample a batch of rows with the given conditions.
+
+        This will enter a reject-sampling loop in which rows will be sampled until
+        all of them are valid and match the requested conditions. If `max_tries`
+        is exceeded, it will return as many rows as it has sampled, which may be less
+        than the target number of rows.
+
+        Input conditions is taken both in the raw input format, which will be used
+        for filtering during the reject-sampling loop, and already transformed
+        to the model format, which will be passed down to the model if it supports
+        conditional sampling natively.
+
+        If condition columns are float values, consider a match anything that is
+        relatively closer than the given ``float_rtol`` and then make the value exact.
+
+        If the model does not have any data columns, the result of this call
+        is a dataframe of the requested length with no columns in it.
+
+        Args:
+            batch_size (int):
+                Number of rows to sample for this batch. If not given the model
+                will generate as many rows as there were in the
+                data passed to the ``fit`` method.
+            max_tries (int):
+                Number of times to retry sampling until the batch size is met.
+                Defaults to 100.
+            conditions (dict):
+                The dictionary of conditioning values in the original input format.
+            transformed_conditions (dict):
+                The dictionary of conditioning values transformed to the model format.
+            float_rtol (float):
+                Maximum tolerance when considering a float match.
+            progress_bar (tqdm.tqdm or None):
+                The progress bar to update when sampling. If None, a new tqdm progress
+                bar will be created.
+            output_file_path (str or None):
+                The file to periodically write sampled rows to. If None, does not write
+                rows anywhere.
+
+        Returns:
+            pandas.DataFrame:
+                Sampled data.
+        """
+        num_rows_to_sample = batch_size
+
+        counter = 0
+        num_valid = 0
+        prev_num_valid = None
+        remaining = batch_size
+        sampled = pd.DataFrame()
+
+        while num_valid < batch_size:
+            if counter >= max_tries:
```

Comment:
Shouldn't we raise a warning if this happens, like we do in CTGAN?

Reply:
@amontanez24 may have the proper answer for this.

Reply:
I don't think `CTGAN` has any different warning messages for sampling. Which one are you referring to?

Reply:
The  warning for not being able to sample enough rows surfaces later down the pipeline